plugins/lookup/consul_kv.py

```diff
@@ -146,7 +147,7 @@ def run(self, terms, variables=None, **kwargs):
         try:
             for term in terms:
                 params = self.parse_params(term)
-                consul_api = consul.Consul(host=host, port=port, scheme=scheme, verify=validate_certs, cert=client_cert)
+                consul_api = consul.Consul(host=host, port=port, scheme=scheme, token=token, verify=validate_certs, cert=client_cert)
```

Comment:
I'm not familiar with the API, but what happens if `token` is provided both here and in the `get()` call? Does the one provided to `get()` override the one provided here?

Reply:
(Depending on how that works, it might make sense to instead modify `parse_params()` below, to use the global options (retrieved with `get_option`) as the default values in `paramvals`. The other parameters there (`recursive`, `index`, `datacanter`) have the same issue as `token`.

Reply:
Well I arrived here after big fiddling with jinja filters in my lookup , all while trying first steps on creating a collection :smile: 
When I saw :  
See https://github.com/ansible-collections/community.general/pull/303 :
* Options handling was switched to get_option approach

...  I thought "my parameter" git overlooked.

But since we are now discussing this publicly: 
* I still have the mostly fresh consul-cluster without production data on it 
  * What can I do produce more information ?

I will try to read and come up with more hands on knowlegde..

Reply:
> (Depending on how that works, it might make sense to instead modify `parse_params()` below, to use the global options (retrieved with `get_option`) as the default values in `paramvals`. The other parameters there (`recursive`, `index`, `datacanter`) have the same issue as `token`.

yes :
* I was afraid forcing token into the call might lead to errors when token is not part of the lookup
  * but that does not seem to be so: https://github.com/fkuep/testing-consul-kv/issues/2#issuecomment-809846132
* also You are right , the other parms [are ignored too](https://github.com/fkuep/testing-consul-kv/issues/3#issuecomment-809868251)

Awaiting feedback!